raiden/channel.py

```diff
@@ -543,6 +543,32 @@ def __init__(self, our_state, partner_state, external_state,
         self.on_task_completed_callbacks = list()  # XXX naming
 
     @property
+    def status(self):
+        status = None
+        if self.isopen:
+            status = 'open'
+            return status
+        else:
+            if self.external_state.settled_block != 0:
+                status = 'settled'
+                if self.external_state.closed_block != 0:
+                    status = 'closed'
+                return status
+        raise Exception('invalid state')
```

Comment:
@ezdac Yes when the channel is `settled`, the `closed_block` is not `0` since you always need to close a channel before you are able to settle it.

> but we should add a comment that the order matters.

What do you mean by that? That the order of the conditional checks in the `if` `elif` `else` matter? That's a given. No need for a comment.

I agree with Konrad's comment and doing the necessary changes at the moment.